docs/en/02_Developer_Guides/14_Files/05_File_Migration.md

```diff
@@ -77,3 +77,24 @@ SilverStripe\Assets\Flysystem\FlysystemAssetStore:
 ```
 
 Note that this will not allow you to utilise certain file versioning features in 4.0.
+
+## Migrating substantial number of files
+
+The time it takes to run the file migration will depend on the number of files and their size. The generation of thumbnail will depend on the number and dimension of your images.
+
+If you are migrating a substantial number of files, you should run file migration task either as a queued job or on the command line. If the migration task fails or times out, you can start it again and will pick off where it left off.
```

Comment:
```suggestion
If you are migrating a substantial number of files, you should run file migration task either as a queued job or on the command line. If the migration task fails or times out, you can start it again and it will pick up where it left off.
```